### PR TITLE
validators using metaprogramming

### PR DESCRIPTION
**Pull request by mayuriv-webonise**

### CODE REVIEW

File: validators.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class DynamicValidator
+  def initialize(attribute, rules)
+    @attribute = attribute
+    @rules = rules
+  end
```
Comment (balakrishna-webonise) [inline validators.rb:6]: Pls add a new line between the methods

File: validators.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  private
+  def define_validation_method(klass, rule)
+    method_name = "validate_#{rule}_of_#{@attribute}"
+    attribute = @attribute.to_sym
+    klass.class_eval do
+      define_method(method_name) do
+        value = send(attribute)
+        case rule
+        when :presence
+          if value.nil? || (value.respond_to?(:empty?) && value.empty?)
```
Comment (balakrishna-webonise) [inline validators.rb:24]: value.blank?

File: validators.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            puts "#{attribute.capitalize} can't be empty"
+          end
+        when :length
+          if value.respond_to?(:length) && value.length < 5
```
Comment (balakrishna-webonise) [inline validators.rb:28]: value&.length < 5

&. - Please read about Ruby safe operator

File: validators.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            puts "#{attribute.capitalize} length should be at least 5 characters"
+          end
+        when :age_limit
+          if attribute == :age && (value.nil? || value < 18)
```
Comment (balakrishna-webonise) [inline validators.rb]: .presence evaluates first right as per your input.
You do not need to check .nil? here again

File: validators.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            puts "Minimum age required is 18 for #{attribute}"
+          end
+        else
+          puts "Validation rule '#{rule}' is not supported"
+        end
+      end
+    end
+  end
+end
```
Comment (balakrishna-webonise) [inline validators.rb:41]: Pls add new line between the classes

File: validators.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+class User
+  attr_accessor :name, :email, :age
```
Comment (balakrishna-webonise) [inline validators.rb:45]: new line. It should be readable

**Review by balakrishna-webonise — CHANGES_REQUESTED** [review]

File: validators.rb (hunk 1 above)
Comment (mayuriv-webonise) [reply validators.rb:6]: added

File: validators.rb (hunk 2 above)
Comment (mayuriv-webonise) [reply validators.rb:24]: added change

Comment (balakrishna-webonise) [reply validators.rb:24]: I do not see the changes for this

**Review by balakrishna-webonise — APPROVED** [review]

**Pull request merged**